Description: Use build flags from environment for menuselect (dpkg-buildflags).
 Necessary for hardening flags.
Author: Simon Ruderich <simon@ruderich.org>
Last-Update: 2012-03-15

Index: asterisk-1.8.8.2~dfsg/Makefile
===================================================================
--- asterisk-1.8.8.2~dfsg.orig/Makefile	2012-03-15 16:55:24.380584542 +0100
+++ asterisk-1.8.8.2~dfsg/Makefile	2012-03-15 16:55:24.392584543 +0100
@@ -884,7 +884,7 @@
 	-@menuselect/nmenuselect menuselect.makeopts && (echo "menuselect changes saved!"; rm -f channels/h323/Makefile.ast main/asterisk) || echo "menuselect changes NOT saved!"
 
 # options for make in menuselect/
-MAKE_MENUSELECT=CC="$(BUILD_CC)" CXX="" LD="" AR="" RANLIB="" CFLAGS="" $(MAKE) -C menuselect CONFIGURE_SILENT="--silent"
+MAKE_MENUSELECT=CC="$(BUILD_CC)" CXX="" LD="" AR="" RANLIB="" CFLAGS="$(CFLAGS)" CPPFLAGS="$(CPPFLAGS)" LDFLAGS="$(LDFLAGS)" $(MAKE) -C menuselect CONFIGURE_SILENT="--silent"
 
 menuselect/menuselect: menuselect/makeopts
 	+$(MAKE_MENUSELECT) menuselect
Index: asterisk-1.8.8.2~dfsg/menuselect/Makefile
===================================================================
--- asterisk-1.8.8.2~dfsg.orig/menuselect/Makefile	2012-03-15 16:53:40.864580603 +0100
+++ asterisk-1.8.8.2~dfsg/menuselect/Makefile	2012-03-15 16:55:24.392584543 +0100
@@ -22,7 +22,7 @@
 
 # Basic set of sources and flags/libraries/includes
 OBJS:=menuselect.o strcompat.o
-CFLAGS:=-g -c -D_GNU_SOURCE -Wall
+CFLAGS+=-g -c -D_GNU_SOURCE -Wall
 
 ifeq ($(MENUSELECT_DEBUG),yes)
   CFLAGS += -DMENUSELECT_DEBUG
@@ -64,14 +64,14 @@
 $(OBJS) $(C_OBJS) $(N_OBJS) $(G_OBJS) $(M_OBJS): autoconfig.h menuselect.h
 
 makeopts autoconfig.h: autoconfig.h.in makeopts.in
-	@./configure $(CONFIGURE_SILENT) CC= LD= AR= CFLAGS=
+	@./configure $(CONFIGURE_SILENT)
 
 $(ALL_TGTS): mxml/libmxml.a
 
 ifdef C_OBJS
 menuselect_curses.o: CFLAGS+=$(C_INCLUDE)
 cmenuselect: $(OBJS) $(C_OBJS)
-	$(CC) -o $@ $^ $(C_LIBS)
+	$(CC) $(LDFLAGS) -o $@ $^ $(C_LIBS)
 else
 cmenuselect:
 endif
@@ -79,7 +79,7 @@
 ifdef G_OBJS
 menuselect_gtk.o: CFLAGS+=$(G_INCLUDE)
 gmenuselect: $(OBJS) $(G_OBJS)
-	$(CC) -o $@ $^ $(G_LIBS)
+	$(CC) $(LDFLAGS) -o $@ $^ $(G_LIBS)
 else
 gmenuselect:
 endif
@@ -87,13 +87,13 @@
 ifdef N_OBJS
 menuselect_newt.o: CFLAGS+=$(N_INCLUDE)
 nmenuselect: $(OBJS) $(N_OBJS)
-	$(CC) -o $@ $^ $(N_LIBS)
+	$(CC) $(LDFLAGS) -o $@ $^ $(N_LIBS)
 else
 nmenuselect:
 endif
 
 menuselect: $(OBJS) $(M_OBJS)
-	$(CC) -o $@ $^ $(M_LIBS)
+	$(CC) $(LDFLAGS) -o $@ $^ $(M_LIBS)
 
 mxml/libmxml.a:
 	@if test ! -f mxml/Makefile ; then cd mxml && ./configure ; fi
